src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs

```diff
@@ -187,6 +184,35 @@ private bool ShouldComplexifyNode(SyntaxNode node, bool isInConflictLambdaBody)
                         node is BaseTypeSyntax);
             }
 
+            public override SyntaxNode VisitAssignmentExpression(AssignmentExpressionSyntax node)
+            {
+
+                const string deconstructMethodName = "Deconstruct";
+                var result = base.VisitAssignmentExpression(node);
+                if (_isProcessingComplexifiedSpans ||
+                    // This is only relevant when renaming to or from Deconstruct().
+                    (_replacementText != deconstructMethodName && _originalText != deconstructMethodName))
+                    return result;
+
+                // Link the = token to the Deconstruct() method to make sure its binding doesn't change.
+                var deconstructSymbols = (_speculativeModel ?? _semanticModel).GetSemanticInfo(
+                        node.OperatorToken, _solution.Workspace, _cancellationToken).ReferencedSymbols;
+                // Even if there aren't any symbols, show a conflict if this rename makes it bind.
+                var renameDeclarationLocations = ConflictResolver.CreateDeclarationLocationAnnotationsAsync(
+                    _solution, deconstructSymbols, _cancellationToken).WaitAndGetResult_CanCallOnBackground(_cancellationToken);
+
+                return _renameAnnotations.WithAdditionalAnnotations(result, new RenameActionAnnotation(
+                    node.OperatorToken.Span,
+                    isRenameLocation: false,
+                    prefix: null,
+                    suffix: null,
+                    renameDeclarationLocations: renameDeclarationLocations,
+                    isOriginalTextLocation: false,
+                    isNamespaceDeclarationReference: false,
+                    isInvocationExpression: true,
```

Comment:
Why does this flag exist?

Under what circumstances would http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Rename/ConflictEngine/ConflictResolver.Session.cs,604 find something when `GetSymbolsTouchingPosition()` doesn't?